### ISSUE TITLE
null props considered differently in getDefaultProps vs. isRequired

### ISSUE THREAD

**Issue by aweary** [facebook/prop-types#110]
Moving from the React repo, original issue: https://github.com/facebook/react/issues/2166

**Comment by jharris4** [comment]
Thanks for linking this issue! Interesting that 39 people reacted with a thumbs up to `+1 for isDefined / isRequired`

There's a lengthy discussion about the topic of null props going on in the https://github.com/facebook/prop-types/pull/90

**Comment by KeitIG** [comment]
For those interested in a workaround, you can always define your own propTypes:

```
// Number or null
foo: function(props, propName, componentName) {
  if (props[propName] !== null || typeof props[propName] !== 'number')
    return new Error(
      'Invalid prop `' + propName + '` supplied to' +
      ' `' + componentName + '`. Validation failed.'
    );
}
```

And it is perfectly possible to make it re-usable as it is a function.

**Comment by virgofx** [comment]
@KeitIG I think the issue is that people prefer an out of box solution to handle such a common use-case.

**Comment by KeitIG** [comment]
I totally agree, I am just leaving this solution in case of someone not knowing how to tackle this in the meantime. (like me, this morning).
